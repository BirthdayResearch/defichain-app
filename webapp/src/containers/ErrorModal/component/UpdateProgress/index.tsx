import React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import styles from '../../errorModal.module.scss';
import { Progress, Button, Row, Col } from 'reactstrap';
import {
  sendUpdateResponse,
  showAvailableUpdateResponse,
} from '../../../../app/update.ipcRenderer';
import {
  closeUpdateApp,
  closePostUpdate,
  closeUpdateAvailable,
} from '../../reducer';
import { UPDATE_MODAL_CLOSE_TIMEOUT } from '../../../../constants';

interface UpdateModalProps {
  isUpdateError: string;
  updateAppinfo: any;
  postUpdateFlag: boolean;
  showUpdateAvailable: boolean;
  isUpdateStarted: boolean;
  closeUpdateApp: () => void;
  closePostUpdate: () => void;
  closeUpdateAvailable: () => void;
}

const UpdateModal: React.FunctionComponent<UpdateModalProps> = (
  props: UpdateModalProps
) => {
  const {
    updateAppinfo,
    showUpdateAvailable,
    postUpdateFlag,
    isUpdateStarted,
    isUpdateError,
    closeUpdateApp,
    closePostUpdate,
    closeUpdateAvailable,
  } = props;

  const percent = Number(updateAppinfo.percent || 0).toFixed(2);

  const closeModal = (fn) => {
    closeUpdateApp();
    setTimeout(fn, UPDATE_MODAL_CLOSE_TIMEOUT);
  };

  if (isUpdateError) {
    return (
      <>
        <p className='text-center'>{isUpdateError}</p>
        <div className='d-flex justify-content-end'>
          <Button className='ml-4' onClick={closeUpdateApp}>
            {I18n.t('alerts.closeBtnLabel')}
          </Button>
        </div>
      </>
    );
  }

  const loadHtml = () => {
    if (showUpdateAvailable) {
      return (
        <>
          <p className='text-center'>
            {I18n.t('alerts.showUpdateAvailableNotice')}
          </p>
          <div className='d-flex justify-content-end'>
            <Button
              color='primary'
              onClick={() => showAvailableUpdateResponse()}
            >
              {I18n.t('alerts.yesShowUpdateAvailableButton')}
            </Button>
            <Button
              className='ml-4'
              onClick={() => closeModal(closeUpdateAvailable)}
            >
              {I18n.t('alerts.noShowUpdateAvailableButton')}
            </Button>
          </div>
        </>
      );
    }

    if (postUpdateFlag) {
      return (
        <>
          <p className='text-center'>{I18n.t('alerts.updateAppNoticeline1')}</p>
          <p className='text-center'>{I18n.t('alerts.updateAppNoticeline2')}</p>
          <div className='d-flex justify-content-end'>
            <Button color='primary' onClick={() => sendUpdateResponse()}>
              {I18n.t('alerts.yesUpdateAppNotice')}
            </Button>
            <Button
              className='ml-4'
              onClick={() => closeModal(closePostUpdate)}
            >
              {I18n.t('alerts.noUpdateAppNotice')}
            </Button>
          </div>
        </>
      );
    }

    if (isUpdateStarted) {
      return (
        <>
          <p className='text-center'>{I18n.t('alerts.downloadingUpdate')}</p>
          <div className={styles.errorModal}>
            <Progress animated color='info' value={percent}>
              {percent} %
            </Progress>
          </div>
        </>
      );
    }

    return <div />;
  };
  return (
    <Row>
      {isUpdateStarted && (
        <Col xs={12}>
          <div className='float-right'>
            <Button size='xs' onClick={closeUpdateApp} color='link'>
              _
            </Button>
          </div>
        </Col>
      )}
      <Col xs={12}>{loadHtml()}</Col>
    </Row>
  );
};

const mapStateToProps = (state) => {
  const {
    isUpdateError,
    updateAppinfo,
    postUpdateFlag,
    showUpdateAvailable,
    isUpdateStarted,
  } = state.errorModal;
  return {
    isUpdateError,
    updateAppinfo,
    postUpdateFlag,
    showUpdateAvailable,
    isUpdateStarted,
  };
};

const mapDispatchToProps = {
  closeUpdateApp,
  closePostUpdate,
  closeUpdateAvailable,
};

export default connect(mapStateToProps, mapDispatchToProps)(UpdateModal);
