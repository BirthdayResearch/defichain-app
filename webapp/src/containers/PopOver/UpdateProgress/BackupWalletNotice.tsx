import React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { ModalBody, ModalFooter, Button } from 'reactstrap';
import {
  backupLoadingStart,
  closeBackupLoading,
  showUpdateAvailable,
} from '../reducer';

interface BackupWalletNoticeProps {
  backupLoadingStart: () => void;
  closeBackupLoading: () => void;
  showUpdateAvailable: () => void;
}

const BackupWalletNotice: React.FunctionComponent<BackupWalletNoticeProps> = (
  props: BackupWalletNoticeProps
) => {
  const { backupLoadingStart, closeBackupLoading, showUpdateAvailable } = props;
  const closing = () => {
    closeBackupLoading();
    showUpdateAvailable();
  };
  return (
    <>
      <ModalBody>
        <h1 className='h4'>{I18n.t('alerts.backupWalletNoticeTitle')}</h1>
        {I18n.t('alerts.backupWalletNotice')}
      </ModalBody>
      <ModalFooter>
        <Button size='sm' color='link' onClick={closing}>
          {I18n.t('alerts.noBackupWalletNotice')}
        </Button>
        <Button size='sm' color='primary' onClick={backupLoadingStart}>
          {I18n.t('alerts.yesBackupWalletNotice')}
        </Button>
      </ModalFooter>
    </>
  );
};
const mapStateToProps = () => {};

const mapDispatchToProps = {
  backupLoadingStart,
  closeBackupLoading,
  showUpdateAvailable,
};

export default connect(mapStateToProps, mapDispatchToProps)(BackupWalletNotice);
