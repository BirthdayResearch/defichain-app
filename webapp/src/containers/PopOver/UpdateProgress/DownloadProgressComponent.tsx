import React from 'react';
import { Progress, ModalBody, Button } from 'reactstrap';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { minimizeDownloadProgressModal } from '../reducer';
import BigNumber from 'bignumber.js';

interface DownloadProgressComponentProps {
  updateAppInfo: any;
  minimizeDownloadProgressModal: () => void;
}

const DownloadProgressComponent = (props: DownloadProgressComponentProps) => {
  const { updateAppInfo, minimizeDownloadProgressModal } = props;
  const percent = new BigNumber(updateAppInfo.percent || 0).toFixed(2);
  const minimize = () => minimizeDownloadProgressModal();
  return (
    <>
      <ModalBody>
        <h1 className='h4'>
          {I18n.t('alerts.downloadingUpdate')}
          <Button
            size='xs'
            color='link'
            className='float-right p-0'
            onClick={minimize}
          >
            &minus;
          </Button>
        </h1>
        <p className='text-center'>{percent} %</p>
        <Progress animated color='info' value={percent}></Progress>
      </ModalBody>
    </>
  );
};

const mapStateToProps = ({ popover }) => ({
  updateAppInfo: popover.updateAppInfo,
});

const mapDispatchToProps = {
  minimizeDownloadProgressModal,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DownloadProgressComponent);
