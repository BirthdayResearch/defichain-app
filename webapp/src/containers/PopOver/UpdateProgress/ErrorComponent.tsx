import React from 'react';
import { Button, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { closeUpdateApp } from '../reducer';
import openNewTab from '../../../utils/openNewTab';
import { SITE_DOWNLOADS } from '@defi_types/settings';

interface DownloadProgressComponentProps {
  isUpdateError: string;
  closeUpdateApp: () => void;
}

const DownloadProgressComponent = (props: DownloadProgressComponentProps) => {
  const { isUpdateError, closeUpdateApp } = props;
  const onDownloadClick = () => {
    openNewTab(SITE_DOWNLOADS);
    closeUpdateApp();
  };
  return (
    <>
      <ModalBody>{isUpdateError}</ModalBody>
      <ModalFooter>
        <Button size='sm' color='link' onClick={onDownloadClick}>
          {I18n.t('alerts.download')}
        </Button>
        <Button size='sm' color='primary' onClick={closeUpdateApp}>
          {I18n.t('alerts.closeBtnLabel')}
        </Button>
      </ModalFooter>
    </>
  );
};

const mapStateToProps = ({ popover }) => ({
  isUpdateError: popover.isUpdateError,
});

const mapDispatchToProps = {
  closeUpdateApp,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DownloadProgressComponent);
