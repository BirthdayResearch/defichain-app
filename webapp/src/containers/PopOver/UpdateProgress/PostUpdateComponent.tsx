import React from 'react';
import { Button, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { sendUpdateResponse } from '../../../app/update.ipcRenderer';
import { closePostUpdate } from '../reducer';

interface PostUpdateComponentProps {
  closePostUpdate: () => void;
  closeModal: (fn: any) => void;
}

const PostUpdateComponent = (props: PostUpdateComponentProps) => {
  const { closePostUpdate, closeModal } = props;
  const closing = () => closeModal(closePostUpdate);
  return (
    <>
      <ModalBody>
        <h1 className='h4'>{I18n.t('alerts.updateAppNoticeTitle')}</h1>
        <p>{I18n.t('alerts.updateAppNotice')}</p>
      </ModalBody>
      <ModalFooter>
        <Button size='sm' color='link' onClick={closing}>
          {I18n.t('alerts.restartLaterNotice')}
        </Button>
        <Button size='sm' color='primary' onClick={() => sendUpdateResponse()}>
          {I18n.t('alerts.restartNowNotice')}
        </Button>
      </ModalFooter>
    </>
  );
};

const mapDispatchToProps = {
  closePostUpdate,
};

export default connect(null, mapDispatchToProps)(PostUpdateComponent);
