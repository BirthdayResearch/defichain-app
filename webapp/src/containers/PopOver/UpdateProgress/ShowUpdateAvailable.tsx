import React from 'react';
import { Button, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { showAvailableUpdateResponse } from '../../../app/update.ipcRenderer';
import { closeUpdateAvailable } from '../reducer';

interface ShowUpdateAvailableComponentProps {
  closeUpdateAvailable: () => void;
  closeModal: (fn: any) => void;
}

const ShowUpdateAvailableComponent = (
  props: ShowUpdateAvailableComponentProps
) => {
  const { closeUpdateAvailable, closeModal } = props;
  const closing = () => closeModal(closeUpdateAvailable);
  return (
    <>
      <ModalBody>
        <h1 className='h4'>{I18n.t('alerts.showUpdateAvailableHeader')}</h1>
        {I18n.t('alerts.showUpdateAvailableNotice')}
      </ModalBody>
      <ModalFooter>
        <Button size='sm' color='link' onClick={closing}>
          {I18n.t('alerts.noShowUpdateAvailableButton')}
        </Button>
        <Button size='sm' color='primary' onClick={showAvailableUpdateResponse}>
          {I18n.t('alerts.yesShowUpdateAvailableButton')}
        </Button>
      </ModalFooter>
    </>
  );
};

const mapDispatchToProps = {
  closeUpdateAvailable,
};

export default connect(null, mapDispatchToProps)(ShowUpdateAvailableComponent);
