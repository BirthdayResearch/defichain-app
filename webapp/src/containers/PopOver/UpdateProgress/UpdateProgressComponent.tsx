import React from 'react';
import { connect } from 'react-redux';
import { closeUpdateApp } from '../reducer';
import { UPDATE_MODAL_CLOSE_TIMEOUT } from '../../../constants';
import ErrorComponent from './ErrorComponent';
import ShowUpdateAvailableComponent from './ShowUpdateAvailable';
import PostUpdateComponent from './PostUpdateComponent';
import DownloadProgressComponent from './DownloadProgressComponent';
import Backupwalletnotice from './BackupWalletNotice';

interface UpdateModalProps {
  isUpdateError: string;
  postUpdateFlag: boolean;
  showUpdateAvailable: boolean;
  isUpdateStarted: boolean;
  closeUpdateApp: () => void;
  backupWalletIsOpen: boolean;
}

const UpdateModal: React.FunctionComponent<UpdateModalProps> = (
  props: UpdateModalProps
) => {
  const {
    showUpdateAvailable,
    postUpdateFlag,
    isUpdateStarted,
    isUpdateError,
    closeUpdateApp,
    backupWalletIsOpen,
  } = props;

  const closeModal = (fn) => {
    closeUpdateApp();
    setTimeout(fn, UPDATE_MODAL_CLOSE_TIMEOUT);
  };

  const loadHtml = () => {
    if (isUpdateError) return <ErrorComponent />;

    if (showUpdateAvailable)
      return <ShowUpdateAvailableComponent closeModal={closeModal} />;

    if (postUpdateFlag) return <PostUpdateComponent closeModal={closeModal} />;

    if (isUpdateStarted) return <DownloadProgressComponent />;

    if (backupWalletIsOpen) return <Backupwalletnotice />;

    return <div />;
  };

  return <>{loadHtml()}</>;
};

const mapStateToProps = (state) => {
  const {
    isUpdateError,
    postUpdateFlag,
    showUpdateAvailable,
    isUpdateStarted,
    backupWalletIsOpen,
  } = state.popover;
  return {
    isUpdateError,
    postUpdateFlag,
    showUpdateAvailable,
    isUpdateStarted,
    backupWalletIsOpen,
  };
};

const mapDispatchToProps = {
  closeUpdateApp,
};

export default connect(mapStateToProps, mapDispatchToProps)(UpdateModal);
