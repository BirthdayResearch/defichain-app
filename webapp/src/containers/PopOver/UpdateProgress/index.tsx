import React from 'react';
import { Modal } from 'reactstrap';
import UpdateProgressComponent from './UpdateProgressComponent';
import styles from '../popOver.module.scss';
import { connect } from 'react-redux';

interface UpdateProgressModalProps {
  isUpdateModalOpen: boolean;
}

const UpdateProgressModal = (props: UpdateProgressModalProps) => {
  return (
    <Modal
      isOpen={props.isUpdateModalOpen}
      centered
      contentClassName={styles.onContentModal}
    >
      <UpdateProgressComponent />
    </Modal>
  );
};

const mapStateToProps = ({ popover }) => ({
  isUpdateModalOpen: popover.isUpdateModalOpen,
});

export default connect(mapStateToProps)(UpdateProgressModal);
