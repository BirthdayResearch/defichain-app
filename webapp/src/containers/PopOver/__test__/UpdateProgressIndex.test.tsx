import React from 'react';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import Enzyme from '../../../utils/testUtils/enzyme';
import UpdateProgress from '../UpdateProgress';

describe('UpdateProgress Index', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <UpdateProgress />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
