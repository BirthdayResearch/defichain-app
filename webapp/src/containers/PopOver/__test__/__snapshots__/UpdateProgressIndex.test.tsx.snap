// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProgress Index should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(UpdateProgressModal)>
    <UpdateProgressModal
      dispatch={[Function]}
      isUpdateModalOpen={false}
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
          Object {
            "mountOnEnter": true,
            "timeout": 150,
          }
        }
        centered={true}
        container="body"
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
          Object {
            "timeout": 300,
          }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        trapFocus={false}
        unmountOnClose={true}
        zIndex={1050}
      />
    </UpdateProgressModal>
  </Connect(UpdateProgressModal)>
</Provider>
`;
